Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -532,34 +571,27 @@ private async void OnObjectEditorPropertiesChanged (object sender, NotifyCollect
 			tcs.SetResult (true);
 		}
 
-		private Task<IReadOnlyCollection<PropertyVariation>[]> GetVariationsAsync (IPropertyInfo property)
+		private async Task<IReadOnlyCollection<PropertyVariation>> GetVariationsAsync (IPropertyInfo property)
 		{
 			var variantTasks = new List<Task<IReadOnlyCollection<PropertyVariation>>> (ObjectEditors.Count);
 			for (int i = 0; i < ObjectEditors.Count; i++) {
 				variantTasks.Add (ObjectEditors[i].GetPropertyVariantsAsync (property));
 			}
 
-			return Task.WhenAll (variantTasks);
+			return (await Task.WhenAll (variantTasks)).SelectMany (vs => vs).Distinct ().ToList ();
 		}
 
-		private async Task<IReadOnlyList<PropertyViewModel>> GetViewModelsAsync (IPropertyInfo property, IEnumerable<PropertyVariation> variantsToSkip = null)
+		private async Task<IReadOnlyList<PropertyViewModel>> GetViewModelsAsync (IPropertyInfo property)
 		{
-			PropertyViewModel baseVm = GetViewModel (property);
-			List<PropertyViewModel> vms = new List<PropertyViewModel> ();
-			vms.Add (baseVm);
-
-			HashSet<PropertyVariation> skipped =
-				(variantsToSkip != null) ? new HashSet<PropertyVariation> (variantsToSkip) : null;
+			PropertyViewModel baseVm = CreateViewModel (property);
+			var vms = new List<PropertyViewModel> { baseVm };
 
 			if (baseVm.HasVariations) {
 				using (await AsyncWork.RequestAsyncWork (this)) {
 					var variants = await GetVariationsAsync (property);
-					for (int i = 0; i < variants.Length; i++) {
-						foreach (PropertyVariation variant in variants[i]) {
-							if (skipped != null && skipped.Contains (variant))
-								continue;
-
-							vms.Add (GetViewModel (property, variant));
+					if (baseVm.HasVariantChildren = variants.Count > 0) {
+						foreach (PropertyVariation variant in variants) {
```

Comment:
I think that's `vms.AddRange (variants.Select (v => CreateViewModel (property, v)))`.